frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -294,18 +364,63 @@ pub mod pallet {
 						pool_id_and_state.state == State::Normal,
 						Error::<T>::TransferringInProgress
 					);
-					Pools::<T>::mutate(asset_id, |_| PoolState { pool_id, state: State::Normal });
+					Pools::<T>::mutate(asset_id, |pool| {
+						*pool = Some(PoolState { pool_id, state: State::Transferring });
+					});
+					Self::deposit_event(Event::OccuredFundsTransferring {
+						old_pool_id: pool_id_and_state.pool_id,
+						new_pool_id: pool_id,
+					});
+					Self::transferring_funds_from_old_pool_to_new(
+						asset_id,
+						pool_id_and_state.pool_id,
+						pool_id,
+					)?;
 				},
 				Err(_) => Pools::<T>::insert(asset_id, PoolState { pool_id, state: State::Normal }),
 			}
 			Ok(())
 		}
+		/// Called when an existing asset_id-pool_id pair is replaced by new pool_id.
+		/// Occurs a withdrawal of all funds from the old pool to the Vault, and then from the Vault
+		/// to the new pool.
+		#[transactional]
+		fn transferring_funds_from_old_pool_to_new(
+			asset_id: T::AssetId,
+			old_pool_id: T::PoolId,
+			new_pool_id: T::PoolId,
+		) -> DispatchResult {
+			AssociatedVaults::<T>::try_mutate(|vaults| {
+				// TODO(belousm): Ask Kevin about uniquest of vaults with underlying asset
+				for vault_id in vaults.iter() {
+					if asset_id == T::Vault::asset_id(vault_id)? {
+						if Self::do_transferring_funds_from_old_pool_to_new(vault_id, old_pool_id)
+							.is_ok()
+						{
+							Self::deposit_event(Event::TransfferedFundsCorrespondedToVault {
+								vault_id: *vault_id,
+								pool_id: new_pool_id,
+							});
+						} else {
+							Self::deposit_event(Event::UnableToTransfferFundsCorrespondedToVault {
+								vault_id: *vault_id,
+								pool_id: new_pool_id,
+							});
+						}
+					}
+				}
+				Pools::<T>::mutate(asset_id, |_| PoolState {
+					pool_id: new_pool_id,
+					state: State::Normal,
+				});
```

Comment:
As you told me early it won't work, you should mutate the value in a closure.

Reply:
Resolved!